src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -357,4 +405,139 @@ public function bulkDelete(array $productIds): void
             sprintf('Failed to delete following products: "%s"', implode(', ', $failedIds))
         );
     }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $shopId
+     *
+     * @return bool
+     */
+    public function isAssociatedToShop(ProductId $productId, ShopId $shopId): bool
+    {
+        return $this->hasShopAssociation($productId->getValue(), Product::class, $shopId);
+    }
+
+    /**
+     * @param ProductId $productId
+     *
+     * @return ShopId[]
+     */
+    public function getAssociatedShopIds(ProductId $productId): array
+    {
+        $qb = $this->connection->createQueryBuilder();
```

Comment:
If this is a part of the Adapter, wouldn't be better to just use `ObjectModel::getAssociatedShops`?

Reply:
I didn't know these methods existed in the ObjectModel, thanks I'll refacto the abstract class a bit to rely on them!

Reply:
For the same reason as your comment related to `isAssociatedToShop` using `getAssociatedShops` required the legacy object to be instanciated to be used.

Form now I admit that this method is only used in `partialUpdateForShopConstraint` which uses an instance of `Product` but this method will likely be useful in other uses cases where we need to get the list of associated shops for a product for example. 

I admit this new kind of repositories are a bit confusing because they mix doctrine queries and use of ObjectModel, also not all of them use the Doctrine connection, so I can't even integrate the connection and dbPrefix in the abstract class.

I think it would be worth thinking of a way to distinct and combine these two kinds of repositories for cleaner implementation, but that would be the purpose of another PR.